presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduSplitManager.java

```diff
@@ -45,10 +60,26 @@ public ConnectorSplitSource getSplits(
             ConnectorTableHandle table,
             SplitSchedulingStrategy splitSchedulingStrategy,
             DynamicFilter dynamicFilter)
+    {
+        if (dynamicFilter.isComplete() || !dynamicFilter.isAwaitable()) {
+            return getSplitSource(table, splitSchedulingStrategy, dynamicFilter);
+        }
+        long blockingTimeoutMillis = getDynamicFilteringProbeBlockingTimeout(session).toMillis();
+        if (blockingTimeoutMillis <= 0) {
```

Comment:
Can be simplified by removing the conditions above and modifying this to `if (blockingTimeoutMillis <= 0 || !dynamicFilter.isAwaitable())`

Reply:
Modified